clang/include/clang/AST/DeclCXX.h

```diff
@@ -2855,6 +2855,11 @@ class CXXDestructorDecl : public CXXMethodDecl {
     return getCanonicalDecl()->OperatorDeleteThisArg;
   }
 
+  /// Will this destructor ever be called when considering which deallocation
+  /// function is associated with the destructor? Can optionally be passed an
+  /// 'operator delete' function declaration to test against specifically.
+  bool isDestructorCalled(const FunctionDecl *OpDel = nullptr) const;
```

Comment:
For readability of call sites, the name of this function should probably mention `delete` in some way. (`isCalledByDelete` maybe?)

Reply:
Good call, I've renamed, thanks!